examples/callback/example_callback.cpp

```diff
@@ -98,13 +98,11 @@ int main(int argc, char *argv[])
     Kokkos::View<int *, MemorySpace> offsets("offsets", 0);
     bvh.query(ExecutionSpace{}, FirstOctant{}, PrintfCallback{}, values,
               offsets);
-#ifndef __NVCC__
+#if !defined(__NVCC__) && !defined(KOKKOS_ENABLE_SYCL)
```

Comment:
Does SYCL not support generic lambdas or are you doing this because the lambda would be no-op?

Reply:
I disabled it since printf just doesn't (yet?) work on CUDA devices. I think I like guarding with `__SYCL_DEVICE_ONLY__` better, though. I'll change.